src/main/java/net/hypixel/nerdbot/command/UserCommands.java

```diff
@@ -0,0 +1,150 @@
+        }
+
+        Reminder reminder = new Reminder(description, date, event.getChannel().getId(), event.getUser().getId());
+        InsertOneResult result = reminder.save();
+
+        if (result != null && result.wasAcknowledged() && result.getInsertedId() != null) {
+            event.reply("I will remind you at " + new DiscordTimestamp(date.getTime()).toLongDateTime() + " about:")
+                    .addEmbeds(new EmbedBuilder().setDescription(description).build())
+                    .setEphemeral(true)
+                    .queue();
+            reminder.schedule();
+        } else {
+            event.reply("Could not save your reminder, please try again later!").queue();
+            log.error("Could not save reminder: " + reminder + " for user: " + event.getUser().getId() + " (" + result + ")");
+        }
+    }
+
+    @JDASlashCommand(name = "remind", subcommand = "list", description = "View your reminders")
+    public void listReminders(GuildSlashEvent event) {
+        List<Reminder> reminders = NerdBotApp.getBot().getDatabase().findAllDocuments(NerdBotApp.getBot().getDatabase().getCollection("reminders", Reminder.class), Filters.eq("userId", event.getUser().getId())).into(new ArrayList<>());
+
+        // Sort these by newest first
+        reminders.sort((o1, o2) -> {
+            if (o1.getTime().before(o2.getTime())) {
+                return -1;
+            } else if (o1.getTime().after(o2.getTime())) {
+                return 1;
+            } else {
+                return 0;
+            }
+        });
+
+        if (reminders.isEmpty()) {
+            event.reply("You have no reminders!").setEphemeral(true).queue();
+            return;
+        }
+
+        StringBuilder builder = new StringBuilder("**Your reminders:**\n");
+        for (Reminder reminder : reminders) {
+            builder.append("(").append(reminder.getUuid()).append(") ").append(new DiscordTimestamp(reminder.getTime().getTime()).toLongDateTime()).append(" - `").append(reminder.getDescription()).append("`\n");
+        }
+
+        builder.append("\n**You can delete reminders with `/remind delete <uuid>`**");
+        event.reply(builder.toString()).setEphemeral(true).queue();
+    }
+
+    @JDASlashCommand(name = "remind", subcommand = "delete", description = "Delete a reminder")
+    public void deleteReminder(GuildSlashEvent event, @AppOption String uuid) {
+        UUID parsed = UUID.fromString(uuid);
+        Reminder reminder = NerdBotApp.getBot().getDatabase().findDocument(NerdBotApp.getBot().getDatabase().getCollection("reminders", Reminder.class), Filters.and(Filters.eq("userId", event.getUser().getId()), Filters.eq("uuid", parsed))).first();
+
+        if (reminder == null) {
+            event.reply("Could not find reminder: " + uuid).setEphemeral(true).queue();
+            return;
+        }
+
+        if (reminder.getTimer() != null) {
+            reminder.getTimer().cancel();
+        }
+
+        reminder.delete();
```

Comment:
this method returns a DeleteResult, you should be checking it and responding to the user accordingly